### PR TITLE
Split travis integration tests

### LINKED ISSUE #1089: Proposal: Split integration test into multiple travis jobs
The integration tests are getting longer
There is a max job time of 50m on travis free tier
We should split some of the integration tests into their own travis jobs
Primarily
TestRun and TestLayers
The other tests can run in the same Travis job

My proposal is to have a job which explicitly passes `-run TestRun` and another which passes `-run TestLayers` and a third which uses `go test ./integration/.. -list=Test` to list all the integration tests and filter out `TestRun` and `TestLayers` from that list using `grep`, then pass that as the regex query to `go test`

IE

`go test ./integration/... -run "TestGitBuildcontext|TestCache|TestRelativePaths"`

cc @samos123 @tejal29

### PR DESCRIPTION
**Pull request by cvgw**
Fixes #1089 

**Description**

Split integration tests into three travis jobs and move image building to it's own stage.

TestRun and TestLayers each have their own job while the rest of the integration tests run in a single job

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [ ] Adds integration tests if needed.

_See [the contribution guide](../CONTRIBUTING.md) for more details._

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit tests and or integration tests added.

**Release Notes**

### CODE REVIEW

File: integration/images.go [hunk 1]
```diff
@@ -356,28 +356,45 @@ func (d *DockerFileBuilder) buildCachedImages(config *integrationTestConfig, cac
 }
 
 // buildRelativePathsImage builds the images for testing passing relatives paths to Kaniko
-func (d *DockerFileBuilder) buildRelativePathsImage(imageRepo, dockerfile, serviceAccount string) error {
+func (d *DockerFileBuilder) buildRelativePathsImage(imageRepo, dockerfile, serviceAccount, buildContextPath string) error {
```
Comment (samos123) [inline integration/images.go]: I think your other patch already made this change? probably a rebase needed after your other patch is merged.

File: integration/integration_test.go [hunk 2]
```diff
@@ -383,14 +383,27 @@ func TestCache(t *testing.T) {
...
-	dockerfile := "Dockerfile_test_copy"
+	dockerfile := "Dockerfile_relative_copy"
 
 	t.Run("test_relative_"+dockerfile, func(t *testing.T) {
 		t.Parallel()
-		imageBuilder.buildRelativePathsImage(config.imageRepo, dockerfile, config.serviceAccount)
 
-		dockerImage := GetDockerImage(config.imageRepo, dockerfile)
-		kanikoImage := GetKanikoImage(config.imageRepo, dockerfile)
+		dockerfile = filepath.Join("./dockerfiles", dockerfile)
```
Comment (samos123) [inline integration/integration_test.go]: same for this file, this seems to need a rebase?

File: integration/dockerfiles/Dockerfile_relative_copy [hunk 3]
```diff
@@ -0,0 +1,3 @@
+FROM alpine@sha256:5ce5f501c457015c4b91f91a15ac69157d9b06f1a75cf9107bf2b62e0843983a
```
Comment (samos123) [inline integration/dockerfiles/Dockerfile_relative_copy]: ditto, rebase after other patch is merged?

**Comment by cvgw** [comment]
This pr depends on the other pr that you are referring to. The intention is
that this Pr would be reviewed and merged after the other one has been
completed.

**Pull request merged**